### ISSUE TITLE
Overriding __hash__ for Mapping class raises reportIncompatibleVariableOverride

### ISSUE THREAD

**Issue by runfalk** [microsoft/pyright#5771]
**Describe the bug**
I have the implemented a frozen dictionary type that also implements `__hash__` when I updated from `1.1.317` to `1.1.323` the `__hash__` implementation raises:

```
/t.py
  /t.py:51:9 - error: "__hash__" overrides symbol of same name in class "Mapping"
    Type cannot be assigned to type "None" (reportIncompatibleVariableOverride)
  /t.py:51:9 - error: Instance variable "__hash__" overrides class variable of same name in class "Mapping" (reportIncompatibleVariableOverride)
2 errors, 0 warnings, 0 informations
```

I believe this is a false positive as `Mapping` to my knowledge shouldn't say anything about `__hash__`. I've also tried inheriting from `Hashable` as well but I get the same error.

I have for now suppressed it using `# pyright: ignore [reportIncompatibleVariableOverride]`.

**Code**
```python
import typing as t
from collections.abc import Iterable, Mapping


K = t.TypeVar("K")
V = t.TypeVar("V")


class FrozenMap(t.Generic[K, V], Mapping[K, V]):
    """An immutable dict implementation"""

    __slots__ = ("_dict", "_hash")
    _dict: dict[K, V]
    _hash: int | None

    @t.overload
    def __init__(self) -> None:
        ...

    @t.overload
    def __init__(self, mapping: Mapping[K, V]) -> None:
        ...

    @t.overload
    def __init__(self, it: Iterable[tuple[K, V]]) -> None:
        ...

    @t.overload
    def __init__(self: "FrozenMap[str, V]", **kwargs: V) -> None:
        ...

    def __init__(self, *args, **kwargs) -> None:  # type: ignore
        self._dict = dict(*args, **kwargs)
        self._hash = None

    def __getitem__(self, key: K) -> V:
        return self._dict[key]

    def __contains__(self, key: t.Any) -> bool:
        return key in self._dict

    def __iter__(self) -> t.Iterator[K]:
        return iter(self._dict)

    def __len__(self) -> int:
        return len(self._dict)

    def __repr__(self) -> str:
        return f"FrozenMap({self._dict!r})"

    def __hash__(self) -> int:
        if self._hash is None:
            self._hash = hash(frozenset(self._dict.items()))
        return self._hash
```

**VS Code extension or command-line**
I'm using the python module to run pyright.

```
$ poetry run pyright --version
pyright 1.1.323
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.324, which I just published. It will also be included in a future release of pylance.
